node/router/src/helpers/resolver.rs

```diff
@@ -13,15 +13,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use parking_lot::RwLock;
 use std::{collections::HashMap, net::SocketAddr};
 
 #[derive(Debug)]
 pub struct Resolver {
     /// The map of the listener address to (ambiguous) peer address.
-    from_listener: RwLock<HashMap<SocketAddr, SocketAddr>>,
+    from_listener: HashMap<SocketAddr, SocketAddr>,
     /// The map of the (ambiguous) peer address to listener address.
-    to_listener: RwLock<HashMap<SocketAddr, SocketAddr>>,
+    to_listener: HashMap<SocketAddr, SocketAddr>,
 }
 
 impl Default for Resolver {
```

Comment:
Minor nit unrelated to your PR, but this is a good opportunity to clean up. Can we derive `Default` in line 19 and then either get rid of `new()` or have new just call `Self::default()`?